tests/v2/utils/library/test_utils.go

```diff
@@ -0,0 +1,322 @@
+func RunMultiThreadTest(testContent commonUtils.TestContent, t *testing.T) {
+
+	commonUtils.LogMessage(fmt.Sprintf("Start Threaded test for %s", testContent.FileName))
+
+	devfileName := testContent.FileName
+	var i int
+	for i = 1; i < numThreads; i++ {
+		testContent.FileName = commonUtils.AddSuffixToFileName(devfileName, "T"+strconv.Itoa(i)+"-")
+		go RunTest(testContent, t)
+	}
+	testContent.FileName = commonUtils.AddSuffixToFileName(devfileName, "T"+strconv.Itoa(i)+"-")
+	RunTest(testContent, t)
+
+	commonUtils.LogMessage(fmt.Sprintf("Sleep 3 seconds to allow all threads to complete : %s", devfileName))
+	time.Sleep(3 * time.Second)
+	commonUtils.LogMessage(fmt.Sprintf("Sleep complete : %s", devfileName))
+
+}
+
+// RunTest : Runs a test to create and verify a devfile based on the content of the specified TestContent
+func RunTest(testContent commonUtils.TestContent, t *testing.T) {
+
+	commonUtils.LogMessage(fmt.Sprintf("Start test for %s", testContent.FileName))
+
+	devfileName := testContent.FileName
+	for i := 1; i <= numDevfiles; i++ {
+
+		testContent.FileName = commonUtils.AddSuffixToFileName(devfileName, strconv.Itoa(i))
+		commonUtils.LogMessage(fmt.Sprintf("Start test for %s", testContent.FileName))
+
+		validator := DevfileValidator{}
+		follower := DevfileFollower{}
+		libraryData, err := devfileData.NewDevfileData("2.0.0")
+		if err != nil {
+			t.Fatalf(commonUtils.LogMessage(fmt.Sprintf("Error creating parser data : %v", err)))
+		}
+		libraryData.SetSchemaVersion("2.0.0")
+		follower.LibraryData = libraryData
+		commonUtils.LogMessage(fmt.Sprintf("Parser data created with schema version : %s", follower.LibraryData.GetSchemaVersion()))
+
+		testDevfile, err := commonUtils.GetDevfile(testContent.FileName, follower, validator)
+		if err != nil {
+			t.Fatalf(commonUtils.LogMessage(fmt.Sprintf("Error creating devfile : %v", err)))
+		}
+
+		testDevfile.RunTest(testContent, t)
+
+		if testContent.EditContent {
+			if len(testContent.CommandTypes) > 0 {
+				err = editCommands(&testDevfile)
+				if err != nil {
+					t.Fatalf(commonUtils.LogErrorMessage(fmt.Sprintf("ERROR editing commands :  %s : %v", testContent.FileName, err)))
+				}
+			}
+			if len(testContent.ComponentTypes) > 0 {
+				err = editComponents(&testDevfile)
+				if err != nil {
+					t.Fatalf(commonUtils.LogErrorMessage(fmt.Sprintf("ERROR editing components :  %s : %v", testContent.FileName, err)))
+				}
+			}
+
+			validator.WriteAndValidate(&testDevfile)
+		}
+	}
+}
+
+// verify verifies the library contents of the specified devfile with the expected content
+func verify(devfile *commonUtils.TestDevfile) error {
+
+	commonUtils.LogInfoMessage(fmt.Sprintf("Verify %s : ", devfile.FileName))
+
+	var errorString []string
+
+	libraryData := devfile.Follower.(DevfileFollower).LibraryData
+	commonUtils.LogInfoMessage(fmt.Sprintf("Get commands %s : ", devfile.FileName))
+	commands, _ := libraryData.GetCommands(common.DevfileOptions{})
```

Comment:
we should not ignore err, if the underlying func is returnin an error, then test should err out. Same for the other calls in this func

Reply:
Fixed